domain-http/interface/src/main/java/org/jboss/as/domain/http/server/ManagementRootConsoleRedirectHandler.java

```diff
@@ -47,15 +46,8 @@ public void handleRequest(HttpServerExchange exchange) throws Exception {
             return;
         }
 
-        String requestUrl = exchange.getRequestURL();
         if (consoleHandler != null && "/".equals(exchange.getRequestPath())) {
-            if (requestUrl.endsWith("/")) {
-                requestUrl = requestUrl.substring(0, requestUrl.length() - 1);
-            }
-            StringBuilder redirect = new StringBuilder(requestUrl);
-            redirect.append(consoleHandler.getContext()).append(consoleHandler.getDefaultPath());
-            exchange.getResponseHeaders().add(Headers.LOCATION, redirect.toString());
-            MOVED_PERMANENTLY.handleRequest(exchange);
+            consoleHandler.getHandler().handleRequest(exchange);
```

Comment:
What happens with ConsoleMode.SLAVE_HC?

Reply:
That behaves the same way as ADMIN_ONLY, only with different URL.

To make sure, I tested this way:
1) Added http management interface into host-slave.xml:
```
<http-interface security-realm="ManagementRealm">                   
    <http-upgrade enabled="true"/>                                  
    <socket interface="management" port="${jboss.management.http.port:9990}"/>                                                                     
</http-interface>
```
2) $ ./domain.sh --host-config=host-slave.xml ...
3) $ curl -s -D - http://123.0.0.1:9990/ 
HTTP/1.1 302 Found
Connection: keep-alive
Location: /consoleerror/noConsoleForSlaveDcError.html

Reply:
Sorry, I misread this and thought it was saying it was giving a 301, which is arguably correct since a slave does not provide a console in normal operation.

But, thinking about it harder I think the 302 is better. A slave is not a slave forever; it can be reconfigured manually and reloaded as a master, and once we do HA DC some slaves will be capable of dynamic switching from slave to master.

Reply:
Moving the status from 301 to 302 only impacts this initial redirect, once the browser receives the redirect the remainder of the caching for things like the console remain in place so the browser only requests those if required.  

Initially the server state was reasonably static in this area so the 301 worked well but with ongoing enhancements we have a variety of situations where this is no longer the case so having the root level redirect happen every time a user attempts to use the console is a good thing to ensure they are using it in relation to the current state of the server.